roles/tests-integration/tests/pool_integration.rs

```diff
@@ -0,0 +1,45 @@
+mod common;
+
+use const_sv2::{
+    MESSAGE_TYPE_COINBASE_OUTPUT_DATA_SIZE, MESSAGE_TYPE_NEW_TEMPLATE,
+    MESSAGE_TYPE_SETUP_CONNECTION, MESSAGE_TYPE_SETUP_CONNECTION_SUCCESS,
+    MESSAGE_TYPE_SET_NEW_PREV_HASH,
+};
+use roles_logic_sv2::{
+    common_messages_sv2::{Protocol, SetupConnection},
+    parsers::{CommonMessages, PoolMessages},
+};
+
+#[tokio::test]
+async fn success_pool_template_provider_connection() {
+    let sniffer_addr = common::get_available_address();
+    let tp_addr = common::get_available_address();
+    let pool_addr = common::get_available_address();
+    let _tp = common::start_template_provider(tp_addr.port()).await;
+    let sniffer = common::start_sniffer(sniffer_addr, tp_addr).await;
+    let _ = common::start_pool(Some(pool_addr), Some(sniffer_addr)).await;
+    match sniffer.next_downstream_message() {
```

Comment:
since this is the first ever test, we will be using it as a reference to write new tests in the future
so it makes sense to add some extra comments to guide us in the future

```suggestion
    // assert message type as well as specific message fields
    match sniffer.next_downstream_message() {
```

Reply:
I wonder if we can encapsulate this match logic into some reusable pattern via macro magic?

just to avoid verbosity inside the tests

just a high level suggestion though, I'm not very good with macros so I'm not sure what would be really possible here

Reply:
First thing which comes in my mind is to have a function like:
`expect_downstream_message_field` which gets `msg_type`, `msg_field`, and `msg_field_value` as parameters, and then it compares the next message got from the queue with the value passed as parameter.

Reply:
> First thing which comes in my mind is to have a function like: `expect_downstream_message_field` which gets `msg_type`, `msg_field`, and `msg_field_value` as parameters, and then it compares the next message got from the queue with the value passed as parameter.

this will probably be hard to achieve since we don't know the type for `msg_field_value` before hand

that's why I believe macro is the only viable option

anyways my suggestion above doesn't need to be implemented on this PR, I can live with the match statements for now

as @jbesraa proposed, we should aim to merge this and move on to https://github.com/stratum-mining/stratum/issues/1121

as we make progress on #1121, we can still come back and make adjustments to #1122

Reply:
Totally agree. To me this one is good to go

Reply:
before merging, I still thing it would be good to add the comments on the suggestion that started this thread

Reply:
Added a macro now to achieve what @GitGab19 mentioned

Reply:
That's amazing, will test it and review this PR again in some hours